src/state_management.c

```diff
@@ -97,21 +107,11 @@ void get_state_str(const struct GameState *game_state, char *str, size_t size) {
                       LIFE_STAGE_STRING[game_state->persistent.stage]);
 
     // Append every individual feature
-    str += copied;
-    size -= copied;
-    copied = get_text_xp(game_state, str, size);
-    str += copied;
-    size -= copied;
-    copied = snprintf(str, size, "\n");
-    str += copied;
-    size -= copied;
-    copied = get_text_hu(game_state, str, size);
-    str += copied;
-    size -= copied;
-    copied = snprintf(str, size, "\n");
-    str += copied;
-    size -= copied;
-    copied = get_text_hp(game_state, str, size);
+		for(uint32_t i = 0; i < NUM_FEAT; ++i) {
+			str += copied;
+			size -= copied;
+			copied = features[i].get_text(game_state, str, size);
```

Comment:
Missing the snprintf()

Reply:
I moved it inside `get_text()` function of each feature